systems/plants/collision/BulletModel.cpp

```diff
@@ -321,125 +433,75 @@ namespace DrakeCollision
-              //END_DEBUG
-              has_result = findClosestPointsBtwBodies(bodyA.getBodyIdx(),
-                      bodyB.getBodyIdx(), active_element_groups,
-                      c);
+    vector<ElementIdPair> id_pairs;
+    const auto elements_end = elements.end();
+    for (auto idA_iter = ids_to_check.begin();
+         idA_iter != ids_to_check.end();
+         ++idA_iter) {
+      auto elementA_iter = elements.find(*idA_iter);
+      if (elementA_iter != elements_end) {
+        for (auto idB_iter = idA_iter+1;
+            idB_iter != ids_to_check.end();
+            ++idB_iter) {
+          auto elementB_iter = elements.find(*idB_iter);
+          if (elementB_iter != elements_end) {
+            if (elements[*idA_iter]->collidesWith(elements[*idB_iter].get())) {
+              id_pairs.push_back(make_pair(*idA_iter, *idB_iter));
             }
           }
         }
-      }
+      } 
     }
-    c->getResults(bodyA_idx, bodyB_idx, ptsA, ptsB,normal,distance);
-    
-    //DEBUG
-    //std::cout << "ModelTemplate:closestPointsAllBodies:" << std::endl;
-    //std::cout << "ptsA:" << std::endl;
-    //std::cout << ptsA.transpose() << std::endl;
-    //std::cout << "ptsB:" << std::endl;
-    //std::cout << ptsB.transpose() << std::endl;
-    //std::cout << "normal:" << std::endl;
-    //std::cout << normal.transpose() << std::endl;
-    //END_DEBUG
-    //DEBUG
-    //} catch (const std::out_of_range& oor) {
-    //std::string msg("In ModelTemplate::closestPointsAllBodies:\n");
-    //throw std::out_of_range(msg + oor.what());
-    //}
-    //END_DEBUG
-    return has_result;
-  };
+    return closestPointsPairwise(id_pairs, use_margins, closest_points);
+  }
+
+  bool BulletModel::closestPointsPairwise(const vector<ElementIdPair>& id_pairs, 
+                                          const bool use_margins,
+                                          vector<PointPair>& closest_points)
+  {
+    unique_ptr<ResultCollector> c(new ResultCollector());
+    for (auto id_pair_iter = id_pairs.begin();
+        id_pair_iter != id_pairs.end();
+        ++id_pair_iter) {
+      findClosestPointsBtwElements(id_pair_iter->first, id_pair_iter->second, 
+                                   use_margins, c);
+    }
+
+    closest_points = c->getResults();
+    return closest_points.size() > 0;
+  }
   
-  bool BulletModel::allCollisions(vector<int>& bodyA_idx,
-          vector<int>& bodyB_idx,
-          MatrixXd& ptsA, MatrixXd& ptsB)
+  bool BulletModel::collisionPointsAllToAll(const bool use_margins,
+                                            vector<PointPair>& collision_points) 
   {
     BulletResultCollector c;
     MatrixXd normals;
     vector<double> distance;
-    bt_collision_world->performDiscreteCollisionDetection();
-    int numManifolds = bt_collision_world->getDispatcher()->getNumManifolds();
+    bullet_world.bt_collision_world->performDiscreteCollisionDetection();
```

Comment:
Looks like this is using bullet_world, regardless of the value of use_margins.  It should probably consult getBulletWorld() first.